log4j-core/src/main/java/org/apache/logging/log4j/core/config/xml/XmlConfiguration.java

```diff
@@ -142,7 +141,7 @@ public XmlConfiguration(final LoggerContext loggerContext, final ConfigurationSo
             try (final InputStream is =
                     Loader.getResourceAsStream(schemaResource, XmlConfiguration.class.getClassLoader())) {
                 if (is != null) {
-                    final javax.xml.transform.Source src = new StreamSource(is, LOG4J_XSD);
+                    final javax.xml.transform.Source src = new StreamSource(is);
```

Comment:
```suggestion
                    final javax.xml.transform.Source src = new StreamSource(is, schemaResource);
```

The `systemId` parameter is very useful for debugging (it is included in the exception messages).

**Remark**: this code is only executed when using the [strict XML configuration mode](https://logging.apache.org/log4j/2.x/manual/configuration.html#strict-xml) that you PR makes completely obsolete. The XSD that you generate allows users to validate a "concise XML config" and does not have any support for the "strict XML config".